standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -2689,11 +2690,32 @@ private void addPartitionsInternal(String catName, String dbName,
     List<MPartition> mParts = new ArrayList<>();
     List<List<MPartitionPrivilege>> mPartPrivilegesList = new ArrayList<>();
     List<List<MPartitionColumnPrivilege>> mPartColPrivilegesList = new ArrayList<>();
+    List<MFieldSchema> partitionKeys = table.getPartitionKeys();
+    int[] dateIndices = IntStream.range(0, partitionKeys.size())
+            .filter(i -> partitionKeys.get(i).getType().equalsIgnoreCase(ColumnType.DATE_TYPE_NAME))
```

Comment:
You have to validate partition column types only if this config is set 'HIVE_TYPE_CHECK_ON_INSERT'.
See this for reference: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java#L1719

Reply:
This config `HIVE_TYPE_CHECK_ON_INSERT` seems a client conf to correct the user input.

The reason why we want to check `date` value format is that HMS query `date` value in format of `uuuu-MM-dd` strictly:
https://github.com/apache/hive/blob/5e6ce2a66404ef0267c27f407f14e601e566dfc0/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java#L1368-L1371
And many engines like Spark will guarantee the  `date` value in format of `uuuu-MM-dd`, which is also important when use date related udf on such partition column.

Reply:
@saihemanth-cloudera After a double thought, we think it's not good to change the partition values in `add_partitions`. 
The original motivation to do it was that the unit test `org.apache.hive.hcatalog.pig.AbstractHCatLoaderTest#testDatePartitionPushUp` failed after this patch:
Pig engine add partition values with time data like `dt=2016-07-14 15%3A10%3A15`, and then the query by dt cast data to 'yyyy-MM-dd' since [HIVE-14282](https://issues.apache.org/jira/browse/HIVE-14282), then it could not match PART_NAME and return empty result.
This test will fail in JDO query too if we disable direct sql because JDO query also use PART_NAME filter to speed up query.

To solve this issue, we could add more rules to match PART_NAME, that is if the partition type is `date`, then we use `PART_NAME like dt=2023-10-10%` instead of `PART_NAME = 2023-10-10`.